src/Type/Php/SubstrDynamicReturnTypeExtension.php

```diff
@@ -37,72 +38,73 @@ public function getTypeFromFunctionCall(
 	): ?Type
 	{
 		$args = $functionCall->getArgs();
-		if (count($args) === 0) {
+		if (count($args) < 2) {
 			return null;
 		}
 
-		if (count($args) >= 2) {
```

Comment:
this cleanup regarding count-of-args and the indent change over the whole block could be a separate PR to make this diff more readable

Reply:
I opened https://github.com/phpstan/phpstan-src/pull/3439